### PR TITLE
rafter-cld-code-sfcc

### PR DESCRIPTION
**Pull request by asad-rafter**

### CODE REVIEW

File: cartridges/cloudinary_sg_changes/cartridge/templates/default/checkout/cart/cart.isml [hunk 1]
```diff
@@ -184,7 +184,7 @@
 														style="max-width:100%"
 														alt="${lineItem.product.getImage('small',0).alt}"
 														title="${lineItem.product.getImage('small',0).title}"
-														class="cld-responsive"
+														class="${'isResponsive' in cldProductImage && cldProductImage.isResponsive ? 'cld-responsive' : ''}"
```
Comment (yuval-cloudinary) [inline cartridges/cloudinary_sg_changes/cartridge/templates/default/checkout/cart/cart.isml:187]: This comment applies to this and all other changes about cld-responsive.
1. When using cld-responsive then no 'src' attribute should be defined, only 'data-src'
2. When not using cld-responsive then no 'data-src' attribute should be defined, only 'src'
3. Do not use "'isResponsive' in object". It is enough to check object.isResponsive

File: cartridges/cloudinary_sfra_changes/cartridge/templates/default/product/components/productTileImage.isml [hunk 2]
```diff
@@ -2,8 +2,9 @@
     <a href="${pdict.urls.product}">
         <iscomment>Custom start: add cloudinary product image</iscomment>
         <isif condition="${product.CLDEnabled && product.images && product.images.cloudinaryProductImage}">
-            <isset name="isResponsive" value="${product.images.cloudinaryProductImage.isResponsive}" scope="page" />
-            <img class="tile-image cld-responsive" 
+            <isset name="isResponsive" value="${!product.images.cloudinaryProductImage.isResponsive}" scope="page" />
```
Comment (yuval-cloudinary) [inline cartridges/cloudinary_sfra_changes/cartridge/templates/default/product/components/productTileImage.isml:5]: why was the negation added (!product.images.cloudinaryProductImage.isResponsive) ?

File: cartridges/cloudinary_sg_changes/cartridge/templates/default/product/components/productimages.isml [hunk 3]
```diff
@@ -51,13 +51,11 @@
...
         <img
-            itemprop="image"
             class="primary-image ${pdict.cloudinary && pdict.cloudinary.isEnabled ? ('cld-responsive primary-image-' + pdict.Product.ID) : ''}"
-            <isif condition="${pdict.cloudinary && pdict.cloudinary.isEnabled}">
-                data-src="${imageUrl}"
-            </isif>
+            data-src="${imageUrl}"
             src="${imageUrl}"
             alt="${imageAlt}"
+            itemprop="image"
```
Comment (yuval-cloudinary) [inline cartridges/cloudinary_sg_changes/cartridge/templates/default/product/components/productimages.isml:58]: please preserve lines order, so minimizing the amount of changes in the code

File: cartridges/int_cloudinary_sfra/cartridge/controllers/Tile.js [hunk 4]
```diff
@@ -26,7 +26,7 @@ server.append('Show', cache.applyPromotionSensitiveCache, function (req, res, ne
                 cloudinaryModel.addCloudinaryProductImage(product, productPrimaryImg);
             }
 
-            if (!empty(productPrimaryImg.altText)) {
+            if (!empty(productPrimaryImg) && !empty(productPrimaryImg.altText)) {
```
Comment (yuval-cloudinary) [inline cartridges/int_cloudinary_sfra/cartridge/controllers/Tile.js:29]: please use standard notation: productPrimaryImg && (productPrimaryImg.altText || productPrimaryImg.altText === "")
so we make sure we allow emptying the at text

**Review by yuval-cloudinary — CHANGES_REQUESTED** [review]

File: metadata/cloudinary-metadata.zip
Comment (yuval-cloudinary) [inline metadata/cloudinary-metadata.zip:1]: Thanks for fixing the typos in the label.

Jobs.xml doesn't have "Cloudinary - Import Image Path and Alt Text" - is that on purpose?

There are many changes in the system-objecttype-extensions.xml. Most notable is the removal of extension types Catalog, Category, Product, Library, and SitePreferences - is that on purpose?

File: cartridges/cloudinary_sg_changes/cartridge/templates/default/search/productnav.isml [hunk 5]
```diff
@@ -164,7 +165,7 @@
 						<img alt="${nextProdUrls.imgAlt}"
 							title="${nextProdUrls.imgTitle}"
 							src="${nextProdUrls.imgUrl}"
```
Comment (yuval-cloudinary) [inline cartridges/cloudinary_sg_changes/cartridge/templates/default/search/productnav.isml:167]: please have src only for nonresponsive, and data-src only for responsive. Also, use the standard condition

**Review by yuval-cloudinary — CHANGES_REQUESTED** [review]

File: cartridges/cloudinary_sg_changes/cartridge/templates/default/checkout/cart/cart.isml (hunk 1 above)
Comment (yuval-cloudinary) [reply cartridges/cloudinary_sg_changes/cartridge/templates/default/checkout/cart/cart.isml]: @asad-rafter Please fix according to these. 
1. Setting the 'src' with cld-responsive is nullifying the advantage of cld-responsive, so it should be avoided
2. Setting the data-src without cld-responsive is bloating the HTML with unused attribute
3. For readability and consistency, we should use only a single way to check the attribute

File: cartridges/int_cloudinary_sfra/cartridge/controllers/Tile.js (hunk 4 above)
Comment (yuval-cloudinary) [reply cartridges/int_cloudinary_sfra/cartridge/controllers/Tile.js]: @asad-rafter Please address my concern

File: cartridges/cloudinary_sg_changes/cartridge/templates/default/checkout/components/minilineitems.isml [hunk 6]
```diff
@@ -114,18 +114,20 @@
 							</style>
 						</isif>
 						<isif condition="${cldProductImage && 'url' in cldProductImage}">
```
[the remark this answers is not in the document]
Comment (yuval-cloudinary) [reply cartridges/cloudinary_sg_changes/cartridge/templates/default/checkout/components/minilineitems.isml]: @asad-rafter please confirm the fix

File: cartridges/cloudinary_sg_changes/cartridge/templates/default/checkout/cart/cart.isml (hunk 1 above)
Comment (asad-rafter) [reply cartridges/cloudinary_sg_changes/cartridge/templates/default/checkout/cart/cart.isml]: @yuval-cloudinary, these are outdated, can you please go there to https://github.com/cloudinary/cloudinary_sfcc_site_cartridge/pull/3/files and verify if your comments have not been fixed?

Comment (yuval-cloudinary) [reply cartridges/cloudinary_sg_changes/cartridge/templates/default/checkout/cart/cart.isml]: Thanks, I see it now!

**Pull request merged**